<!DOCTYPE html>
<html>
<head>
    <title>All Tests Results</title>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.6/css/bootstrap.min.css" integrity="sha384-1q8mTJOASx8j1Au+a5WDVnPi2lkFfwwEAa8hDDdjZlpLegxhjVME1fgjWPGmkzs7" crossorigin="anonymous">
</head>
<body>
    <div class="container">
        <div class="row">
            <div class="col-xs-12">
                <h2 class="text-capitalize">All Tests Results</h2>
                <p class='attribute'><strong>Start Time: </strong>2023-04-12 13:12:28</p>
                <p class='attribute'><strong>Duration: </strong>0 ms</p>
                <p class='attribute'><strong>Summary: </strong>Total: 4, Pass: 0, Fail: 1, Error: 3</p>
            </div>
        </div>
        <div class="row">
            <div class="col-xs-12 col-sm-10 col-md-10">
                <table class='table table-hover table-responsive'>
                    <thead>
                        <tr>
                            <th>test.testcase.test_splash.TestSplash</th>
                            <th>Status</th>
                            <th></th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr class='danger'>
                            <td class="col-xs-10">test_login_displayed</td>
                            <td class="col-xs-1">
                                <span class="label label-danger" style="display:block;width:40px;">Fail</span>
                            </td>
                            <td class="col-xs-1">
                                <button class="btn btn-default btn-xs">View</button>
                            </td>
                        </tr>
                        <tr style="display:none;">
                            <td class="col-xs-9" colspan="3"><p style="color:maroon;">AssertionError: Login button not displayed after clicking Start button</p><p style="color:maroon;">Traceback (most recent call last):
  File "C:\Users\USER\PycharmProjects\TestAuto\src\test\testcase\test_splash.py", line 23, in test_login_displayed
    assert is_login_displayed, "Login button not displayed after clicking Start button"
AssertionError: Login button not displayed after clicking Start button
</p>
                            </td>
                        </tr>
                        <tr class='warning'>
                            <td class="col-xs-10">test_start_button_click</td>
                            <td class="col-xs-1">
                                <span class="label label-warning" style="display:block;width:40px;">Error</span>
                            </td>
                            <td class="col-xs-1">
                                <button class="btn btn-default btn-xs">View</button>
                            </td>
                        </tr>
                        <tr style="display:none;">
                            <td class="col-xs-9" colspan="3"><p style="color:maroon;">AttributeError: 'NoneType' object has no attribute 'find_element'</p><p style="color:maroon;">Traceback (most recent call last):
  File "C:\Users\USER\PycharmProjects\TestAuto\src\test\testcase\test_splash.py", line 16, in test_start_button_click
    splash.click(PageSplash.btn_start)
  File "C:\Users\USER\PycharmProjects\TestAuto\src\page\page_base.py", line 13, in click
    WebDriverWait(self.driver, 10).until(EC.visibility_of_element_located(by_locator)).click()
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\selenium\webdriver\support\wait.py", line 86, in until
    value = method(self._driver)
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\selenium\webdriver\support\expected_conditions.py", line 139, in _predicate
    return _element_if_visible(driver.find_element(*locator))
AttributeError: 'NoneType' object has no attribute 'find_element'
</p>
                            </td>
                        </tr>
                        <tr>
                            <td colspan="3">
                                Total: 2, Pass: 0, Fail: 1, Error: 1 -- Duration: 0 ms
                            </td>
                        </tr>
                    </tbody>
                </table>
            </div>
        </div>
        <div class="row">
            <div class="col-xs-12 col-sm-10 col-md-10">
                <table class='table table-hover table-responsive'>
                    <thead>
                        <tr>
                            <th>setUpClass (test.testcase.test_app_launch</th>
                            <th>Status</th>
                            <th></th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr class='warning'>
                            <td class="col-xs-10">TestAppLaunch)</td>
                            <td class="col-xs-1">
                                <span class="label label-warning" style="display:block;width:40px;">Error</span>
                            </td>
                            <td class="col-xs-1">
                                <button class="btn btn-default btn-xs">View</button>
                            </td>
                        </tr>
                        <tr style="display:none;">
                            <td class="col-xs-9" colspan="3"><p style="color:maroon;">WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: Unable to find an active device or emulator with OS 13. The following are available: R59T30ASVTK (12)
Stacktrace:
UnknownError: An unknown server-side error occurred while processing the command. Original error: Unable to find an active device or emulator with OS 13. The following are available: R59T30ASVTK (12)
    at getResponseForW3CError (C:\Users\nam\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\protocol\errors.js:804:9)
    at asyncHandler (C:\Users\nam\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\protocol\protocol.js:380:37)</p><p style="color:maroon;">Traceback (most recent call last):
  File "C:\Users\USER\PycharmProjects\TestAuto\src\config\webdriver_setup.py", line 29, in setUpClass
    cls.driver = webdriver.Remote(appium_url, desired_caps)
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\appium\webdriver\webdriver.py", line 234, in __init__
    super().__init__(
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 286, in __init__
    self.start_session(capabilities, browser_profile)
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\appium\webdriver\webdriver.py", line 324, in start_session
    response = self.execute(RemoteCommand.NEW_SESSION, w3c_caps)
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: Unable to find an active device or emulator with OS 13. The following are available: R59T30ASVTK (12)
Stacktrace:
UnknownError: An unknown server-side error occurred while processing the command. Original error: Unable to find an active device or emulator with OS 13. The following are available: R59T30ASVTK (12)
    at getResponseForW3CError (C:\Users\nam\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\protocol\errors.js:804:9)
    at asyncHandler (C:\Users\nam\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\protocol\protocol.js:380:37)
</p>
                            </td>
                        </tr>
                        <tr>
                            <td colspan="3">
                                Total: 1, Pass: 0, Error: 1 -- Duration: 0 ms
                            </td>
                        </tr>
                    </tbody>
                </table>
            </div>
        </div>
        <div class="row">
            <div class="col-xs-12 col-sm-10 col-md-10">
                <table class='table table-hover table-responsive'>
                    <thead>
                        <tr>
                            <th>setUpClass (test.testcase.test_splash2</th>
                            <th>Status</th>
                            <th></th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr class='warning'>
                            <td class="col-xs-10">TestSplash2)</td>
                            <td class="col-xs-1">
                                <span class="label label-warning" style="display:block;width:40px;">Error</span>
                            </td>
                            <td class="col-xs-1">
                                <button class="btn btn-default btn-xs">View</button>
                            </td>
                        </tr>
                        <tr style="display:none;">
                            <td class="col-xs-9" colspan="3"><p style="color:maroon;">WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: Unable to find an active device or emulator with OS 13. The following are available: R59T30ASVTK (12)
Stacktrace:
UnknownError: An unknown server-side error occurred while processing the command. Original error: Unable to find an active device or emulator with OS 13. The following are available: R59T30ASVTK (12)
    at getResponseForW3CError (C:\Users\nam\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\protocol\errors.js:804:9)
    at asyncHandler (C:\Users\nam\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\protocol\protocol.js:380:37)</p><p style="color:maroon;">Traceback (most recent call last):
  File "C:\Users\USER\PycharmProjects\TestAuto\src\config\webdriver_setup.py", line 29, in setUpClass
    cls.driver = webdriver.Remote(appium_url, desired_caps)
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\appium\webdriver\webdriver.py", line 234, in __init__
    super().__init__(
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 286, in __init__
    self.start_session(capabilities, browser_profile)
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\appium\webdriver\webdriver.py", line 324, in start_session
    response = self.execute(RemoteCommand.NEW_SESSION, w3c_caps)
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 440, in execute
    self.error_handler.check_response(response)
  File "C:\Users\USER\PycharmProjects\TestAuto\venv\lib\site-packages\selenium\webdriver\remote\errorhandler.py", line 245, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: Unable to find an active device or emulator with OS 13. The following are available: R59T30ASVTK (12)
Stacktrace:
UnknownError: An unknown server-side error occurred while processing the command. Original error: Unable to find an active device or emulator with OS 13. The following are available: R59T30ASVTK (12)
    at getResponseForW3CError (C:\Users\nam\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\protocol\errors.js:804:9)
    at asyncHandler (C:\Users\nam\AppData\Local\Programs\Appium Server GUI\resources\app\node_modules\appium\node_modules\appium-base-driver\lib\protocol\protocol.js:380:37)
</p>
                            </td>
                        </tr>
                        <tr>
                            <td colspan="3">
                                Total: 1, Pass: 0, Error: 1 -- Duration: 0 ms
                            </td>
                        </tr>
                    </tbody>
                </table>
            </div>
        </div>
    </div>
    <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.4/jquery.min.js"></script>
    <script type="text/javascript">
        $(document).ready(function(){
            $('td').on('click', '.btn', function(e){
                e.preventDefault();
                e.stopImmediatePropagation();
                var $this = $(this);
                var $nextRow = $this.closest('tr').next('tr');
                $nextRow.slideToggle("fast");
                $this.text(function(i, text){
                    if (text === 'View') {
                        return 'Hide';
                    } else {
                        return 'View';
                    };
                });
            });
        });
    </script>
</body>
</html